### ISSUE TITLE
some segments not showing in tmux on macOS

### ISSUE THREAD

**Issue by kadaliao** [powerline/powerline#2191]
Hi there,

I just set up a new mac book and some of the segments are not showing in tmux.

my configuration: ~/.config/powerline/themes/tmux/default.json

```json
{
  "segments": {
    "right": [
      {
        "function": "powerline.segments.common.sys.cpu_load_percent",
        "priority": 30,
        "threshold": 5,
        "before": "CPU: "
      },
      {
        "function": "powerline.segments.common.env.virtualenv",
        "priority": 20
      },
      {
        "function": "powerlinemem.mem_usage.mem_usage_percent",
        "priority": 50,
        "args": {
          "format": "Mem: %d%%"
        }
      },
      {
        "function": "powerline.segments.common.net.network_load"
      },
      {
        "function": "powerline.segments.common.time.date"
      },
      {
        "function": "powerline.segments.common.time.date",
        "name": "time",
        "args": {
          "format": "%H:%M",
          "istime": true
        }
      }
    ]
  }
}
```

But network_load, virtualenv, cpu_load_percent is not showing. 

![image](https://user-images.githubusercontent.com/7401626/147872993-c5d696f9-c3d3-43a7-94c5-08402167bdb4.png)

![image](https://user-images.githubusercontent.com/7401626/147873195-5f6ea086-5179-4432-8474-2a32487095bb.png)

![image](https://user-images.githubusercontent.com/7401626/147873221-28e5c548-98f0-4592-8e3a-0f2be91ac031.png)

The same configuration works fine on other macOS computers.

**Comment by PH111P** [comment]
Hi @kadaliao, thanks for your detailed report.

Some segments require extra python libraries. In particular, the segments you mentioned require:
* network_load: `psutil` or the presence of `/sys/class/net/{interface}/statistics/{rx,tx}_bytes`
* cpu_load_percent: again `psutil`
* virtualenv: This one should be fine as is, it reads the environment variable `VIRTUAL_ENV` though, which could get set incorrectly

Having said that, psutil seems to be installed on your machine (otherwise the segments wouldn't work in the python console either). I could imagine that the tmux renderer has some problems with the threaded segments (the load segments)—Are you using the same powerline version on your machine as the confirmed working installations?
Also, I am not familiar with the `mem_usage` segment; it might be doing something weird.

**Comment by kadaliao** [comment]
@PH111P thanks for the reply.

I'm using powerline version 2.7. (all is installed by `pwoerline-status`)

the virtualenv segment is not showing even the VIRTUAL_ENV variable is set and generated by `powerline tmux right`

![image](https://user-images.githubusercontent.com/7401626/147880972-380246f6-c45a-4086-bad9-992ab7f0635d.png)

**Comment by PH111P** [comment]
Very strange. Just to make sure, that segment is not cut off due to a lack of horizontal space, right? My next suspicion would be that the tmux line never "sees" the updated environment variable and hence never shows the corresponding segment. (This used to be a problem for the cwd segment until it got fixed in https://github.com/powerline/powerline/pull/2127, so perhaps a similar fix is needed here as well)

**Comment by kadaliao** [comment]
@PH111P nope, i made the window small just for ease of screenshot. Good news is that all works fine now after I updated macOS to Monterey 12.1
